rebac-admin-backend/v1/interfaces/groups.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package interfaces
+
+import (
+	"context"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// GroupsService defines an abstract backend to handle Groups related operations.
+type GroupsService interface {
+
+	// ListGroups returns a page of Group objects of at least `size` elements if available.
+	ListGroups(ctx context.Context, params *resources.GetGroupsParams) (*resources.Groups, error)
+	// CreateGroup creates a single Group.
+	CreateGroup(ctx context.Context, group *resources.Group) (*resources.Group, error)
+
+	// GetGroup returns a single Group.
+	GetGroup(ctx context.Context, groupId string) (*resources.Group, error)
```

Comment:
suggestion(proxied by Ales): a previous suggestion by Ales would have us append the string "identified by `groupId`", which I think it's better and I applied it to my godocs